File: CodeSolutions/48. Rotate Image/48.py
// Language: Java
// Commit Timestamp: 2019-12-11T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

// Language: Java
// Commit Timestamp: 2019-12-11T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

// Language: Java
// Commit Timestamp: 2019-12-11T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

class Solution:
  def rotate(self, matrix: list[list[int]]) -> None:
    matrix.reverse()

    for i in range(len(matrix)):
      for j in range(i + 1, len(matrix)):
        matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

File: CodeSolutions/48. Rotate Image/48.cpp
// Language: Java
// Commit Timestamp: 2020-01-10T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

// Language: Java
// Commit Timestamp: 2020-01-10T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

// Language: Java
// Commit Timestamp: 2020-01-10T00:00:00+05:30
// Solution Logic: Placeholder for explanation.

class Solution {
 public:
  void rotate(vector<vector<int>>& matrix) {
    ranges::reverse(matrix);
    for (int i = 0; i < matrix.size(); ++i)
      for (int j = i + 1; j < matrix.size(); ++j)
        swap(matrix[i][j], matrix[j][i]);
  }
};
